<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SGPU_InteractiveParticleSimulationClass</class>
 <widget class="QMainWindow" name="SGPU_InteractiveParticleSimulationClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1264</width>
    <height>892</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SGPU_InteractiveParticleSimulation</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="toolFrame" native="true">
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <property name="leftMargin">
         <number>6</number>
        </property>
        <property name="topMargin">
         <number>6</number>
        </property>
        <property name="rightMargin">
         <number>6</number>
        </property>
        <property name="bottomMargin">
         <number>6</number>
        </property>
        <property name="spacing">
         <number>6</number>
        </property>
        <item row="4" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>320</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="updateButton">
          <property name="text">
           <string>Update</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="startButton">
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Visual Properties</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <property name="leftMargin">
            <number>8</number>
           </property>
           <property name="topMargin">
            <number>8</number>
           </property>
           <property name="rightMargin">
            <number>8</number>
           </property>
           <property name="bottomMargin">
            <number>8</number>
           </property>
           <property name="spacing">
            <number>8</number>
           </property>
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Particle Color:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="colorButton">
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Particle Size:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="particleSize">
               <property name="minimum">
                <double>0.100000000000000</double>
               </property>
               <property name="maximum">
                <double>1.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>0.500000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="stopButton">
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QViewport" name="viewport" native="true"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1264</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QViewport</class>
   <extends>QWidget</extends>
   <header location="global">QViewport.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="SGPU_InteractiveParticleSimulation.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>SGPU_InteractiveParticleSimulationClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>445</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>updateButton</sender>
   <signal>released()</signal>
   <receiver>SGPU_InteractiveParticleSimulationClass</receiver>
   <slot>onUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>445</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colorButton</sender>
   <signal>released()</signal>
   <receiver>SGPU_InteractiveParticleSimulationClass</receiver>
   <slot>onColorButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>382</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>445</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>released()</signal>
   <receiver>SGPU_InteractiveParticleSimulationClass</receiver>
   <slot>onStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>445</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>released()</signal>
   <receiver>SGPU_InteractiveParticleSimulationClass</receiver>
   <slot>onStop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>445</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onUpdate()</slot>
  <slot>onColorButton()</slot>
  <slot>onStart()</slot>
  <slot>onStop()</slot>
 </slots>
</ui>
